extensions/import/src/test/controllers/mainController.test.ts

```diff
@@ -7,25 +7,54 @@ import * as vscode from 'vscode';
 import * as TypeMoq from 'typemoq';
 import { ApiWrapper } from '../../common/apiWrapper';
 import MainController from '../../controllers/mainController';
-import { TestExtensionContext } from '../utils.test';
+import * as constants from '../../common/constants';
+import { TestExtensionContext, getExtensionPath } from '../utils.test';
+import * as fs from 'fs';
+import * as should from 'should';
+import * as path from 'path';
 
 describe('Main Controller', function () {
 	let mockExtensionContext: TypeMoq.IMock<vscode.ExtensionContext>;
 	let mockApiWrapper: TypeMoq.IMock<ApiWrapper>;
+	let extensionPath: string;
+	let flatFileImportServicePath: string;
 
-	this.beforeEach(function () {
-		mockExtensionContext = TypeMoq.Mock.ofType(TestExtensionContext, TypeMoq.MockBehavior.Loose);
+	beforeEach(async function () {
+		extensionPath = await getExtensionPath();
+		flatFileImportServicePath = path.join(await getExtensionPath(), 'flatfileimportservice');
+		// creating a mock Extension Context with current extensionPath
+		mockExtensionContext = TypeMoq.Mock.ofType(TestExtensionContext, TypeMoq.MockBehavior.Loose, true, extensionPath);
 		mockApiWrapper = TypeMoq.Mock.ofType(ApiWrapper);
 	});
 
-	it('Should create new instance successfully', async function () {
-		// mocking createOutputChannel in API wrapper
-		mockApiWrapper.setup(x => x.createOutputChannel(TypeMoq.It.isAny()));
+	it('Should download required binaries and register flatFileImportStartCommand after activate is called', async function () {
+		this.timeout(50000);
 
-		// creating a Main Controller
-		new MainController(mockExtensionContext.object, mockApiWrapper.object);
+		// deleting .net code files to check if activate downloads it again.
+		if(await checkPathExists(flatFileImportServicePath)){
+			await fs.promises.rmdir(flatFileImportServicePath);
+		}
+		should.equal(await checkPathExists(flatFileImportServicePath), false);
 
-		// verifying if the output channel is created
-		mockApiWrapper.verify(x => x.createOutputChannel(TypeMoq.It.isAny()), TypeMoq.Times.once());
+		// using vscode and azdata APIs available during tests
+		mockApiWrapper.callBase = true;
+
+		let mainController = new MainController(mockExtensionContext.object, mockApiWrapper.object);
+
+		await mainController.activate();
+
+		// verifying that the task is registered.
+		mockApiWrapper.verify(x => x.registerTask(constants.flatFileImportStartCommand, TypeMoq.It.isAny()), TypeMoq.Times.once());
+
+		//Checking if .net code files are downloaded
+		should.equal(await checkPathExists(flatFileImportServicePath), true);
 	});
 });
+
+
+async function checkPathExists(path: string): Promise<boolean> {
```

Comment:
I'd suggest putting this into a general utils file instead of here -this is useful for other tests too.